python/tests/unit/arcticdb/version_store/test_parallel.py

```diff
@@ -335,19 +334,18 @@ def test_parallel_overlapping_incomplete_segments(lmdb_version_store, append):
     else:
         lib.write(sym, df_2, parallel=True)
         lib.write(sym, df_1, parallel=True)
-    with pytest.raises(Exception):
+    with pytest.raises(SortingException):
         lib.compact_incomplete(sym, append, False)
 
 
-@pytest.mark.xfail(reason="See https://github.com/man-group/ArcticDB/issues/1249")
 def test_parallel_append_overlapping_with_existing(lmdb_version_store):
     lib = lmdb_version_store
     sym = "test_parallel_append_overlapping_with_existing"
     df_0 = pd.DataFrame({"col": [1, 2]}, index=[pd.Timestamp("2024-01-01"), pd.Timestamp("2024-01-02")])
     lib.write(sym, df_0)
     df_1 = pd.DataFrame({"col": [3, 4]}, index=[pd.Timestamp("2024-01-01T12"), pd.Timestamp("2024-01-03")])
     lib.append(sym, df_1, incomplete=True)
-    with pytest.raises(Exception):
+    with pytest.raises(SortingException):
```

Comment:
Maybe we could add an assert to check the values included in the error message string are not out-by-one or anything? You can `with pytest.raises(...) as exception_info`.

Reply:
I've verified this manually, and any future PR changing these values by 1 should have the reviewers (correctly) asking why.
From a "the tests are the documentation" point of view, I dislike tests that rely on the content of strings, either in logs or in exception messages, as it restricts our ability to modify these strings in the future. The behaviour we are guaranteeing is that this type of error will always raise a `SortingException`, the exception message is an implementation detail that we are free to change, and shouldn't be relied upon by users.